File: src/challenges/java/guessinggame/JSONParser.java
package challenges.java.guessinggame;

import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;
import com.squareup.moshi.Types;
import okio.BufferedSource;
import okio.Okio;

import java.io.File;
import java.io.IOException;
import java.lang.reflect.ParameterizedType;
import java.util.Arrays;
import java.util.List;

public class JSONParser {

    public static void main(String[] args) throws IOException {
        File file = new File("assets/data.json");
        BufferedSource bufferedSource = Okio.buffer(Okio.source(file));
        String jsonString = bufferedSource.readUtf8();
        Moshi moshi = new Moshi.Builder().build();
        ParameterizedType types = Types.newParameterizedType(List.class, QuizWrapper.class);
        JsonAdapter<List<QuizWrapper>> quizWrapperJsonAdapter = moshi.adapter(types);
        List<QuizWrapper> result = quizWrapperJsonAdapter.fromJson(jsonString);
        assert result != null;
        System.out.println(result.get(0).quiz.maths.q2.answer);
    }
}

File: src/challenges/kotlin/jsonparser/JSONParser.kt
package challenges.kotlin.jsonparser

import com.squareup.moshi.JsonAdapter
import com.squareup.moshi.Moshi
import com.squareup.moshi.Types
import okio.Okio
import java.io.File

fun main() {
    val file = File("assets/data.json")
    val source = Okio.source(file)
    val bufferSource = Okio.buffer(source)
    val string = bufferSource.readUtf8()
    val moshi = Moshi.Builder().build()
    val listType = Types.newParameterizedType(List::class.java, QuizWrapper::class.java)
    val jsonAdapter: JsonAdapter<List<Quiz>> = moshi.adapter(listType)
    val result = jsonAdapter.fromJson(string)
    print(result)
}